src/m365/entra/commands/pim/pim-role-assignment-add.ts

```diff
@@ -0,0 +1,319 @@
+      const isAppOnlyAccessToken = accessToken.isAppOnlyAccessToken(token);
+      if (isAppOnlyAccessToken) {
+        throw 'When running with application permissions either userId, userName, groupId or groupName is required';
+      }
+
+      const roleDefinitionId = await this.getRoleDefinitionId(args.options, logger);
+      const principalId = await this.getPrincipalId(args.options, logger);
+
+      if (!args.options.userId && !args.options.userName && !args.options.groupId && !args.options.groupName) {
+        action = 'selfActivate';
+      }
+
+      const requestOptions: CliRequestOptions = {
+        url: `${this.resource}/v1.0/roleManagement/directory/roleAssignmentScheduleRequests`,
+        headers: {
+          'accept': 'application/json;odata.metadata=none'
+        },
+        responseType: 'json',
+        data: {
+          principalId: principalId,
+          roleDefinitionId: roleDefinitionId,
+          directoryScopeId: this.getDirectoryScope(args.options),
+          action: action,
+          justification: args.options.justification,
+          scheduleInfo: {
+            startDateTime: args.options.startDateTime,
+            expiration: {
+              duration: this.getDuration(args.options),
+              endDateTime: args.options.endDateTime,
+              type: this.getExpirationType(args.options)
+            }
+          },
+          ticketInfo: {
+            ticketNumber: args.options.ticketNumber,
+            ticketSystem: args.options.ticketSystem
+          }
+        }
+      };
+
+      const response = await request.post<UnifiedRoleAssignmentScheduleRequest>(requestOptions);
+
+      await logger.log(response);
+    }
+    catch (err: any) {
+      this.handleRejectedODataJsonPromise(err);
+    }
+  }
+
+  private async getRoleDefinitionId(options: Options, logger: Logger): Promise<string> {
+    if (options.roleDefinitionId) {
+      return options.roleDefinitionId;
+    }
+
+    if (this.verbose) {
+      await logger.logToStderr(`Retrieving role definition by its name '${options.roleDefinitionName}'`);
+    }
+
+    const role = await roleDefinition.getRoleDefinitionByDisplayName(options.roleDefinitionName!);
+    return role.id!;
+  }
+
+  private async getPrincipalId(options: Options, logger: Logger): Promise<string> {
+    if (options.userId) {
+      return options.userId;
+    }
+
+    if (options.userName) {
+      if (this.verbose) {
+        await logger.logToStderr(`Retrieving user by its name '${options.userName}'`);
+      }
+
+      return await entraUser.getUserIdByUpn(options.userName);
+    }
+    else if (options.groupId) {
+      return options.groupId;
+    }
+    else if (options.groupName) {
+      if (this.verbose) {
+        await logger.logToStderr(`Retrieving group by its name '${options.groupName}'`);
+      }
+
+      return await entraGroup.getGroupIdByDisplayName(options.groupName);
+    }
+    else {
```

Comment:
Let's remove this last else statement. It's unnecessary and will save some indentation, which will make the code more readable.